test/plugin/base_test.rb

```diff
@@ -1752,6 +1813,16 @@ def latency_conversion
 
   # This module expects the methods below to be overridden.
 
+  # Whether use_grpc is enabled.
+  def use_grpc
```

Comment:
Hmm, would this enable factoring other calls to `assert_prometheus_metric_value` into this file?

Reply:
A bit more refactoring is needed because the prometheus tests needs some error code to be passed into the `setup_logging_stub` method, and the signature is different among REST and gRPC. I'll address that in a separate PR.

Reply:
Hmm, I also wonder whether we could just make the `use_grpc` and `code` parameters to `assert_prometheus_metric_value` optional... That would allow you to get rid of these virtual methods. The values are really irrelevant to what you're trying to check.
One possible implementation:
```ruby
  def assert_prometheus_metric_value(metric_name, expected_value, labels = nil)
    metric = Prometheus::Client.registry.get(metric_name)
    assert_not_nil(metric)
    metric_value = if labels
                     metric.get(labels)
                   else
                     metric.values.values.reduce(0, :+)
                   end
    assert_equal(expected_value, metric_value)
  end
```
Alternatively, you can leave the `labels` default as `{}` and instead check for a special value, e.g., `:all` to aggregate.

Reply:
Does that imply we don't check the `use_grpc` and `code` labels of the metrics anywhere? I thought we do want to verify them since that affects filtering.

I've created b/72162902 and added this note there. Will refactor and combine the test in a separate PR.

Reply:
We will if they're relevant. In the one check you have in the base test, the label values are not -- the success code is always the same (modulo gRPC vs REST), and we assume that `use_grpc` will be set correctly.
I'm not suggesting that you change all calls to `assert_prometheus_metric_value` -- only the two that are in `base_test.rb`, to avoid introducing extra virtual methods `use_grpc` and `ok_status_code`.
FWIW, I felt that, since this PR actually introduces those virtual methods, it's easier to not do that rather than refactor later. WDYT?

Reply:
Ah, I see. Just made labels optional and removed the virtual methods. Will defer combining the prometheus tests till later.

Reply:
Per discussion offline, it makes more sense to explicitly specify `:all` when we want to sum up all metric values regardless of the labels. Just so we can distinguish cases between `get all metrics regardless of the labels` and `get all metrics with empty labels`. Just made the changes accordingly.